services/matrix-provisioning-service/tests/test_admin_client.py

```
@@ -0,0 +1,275 @@
+    )
+
+
+async def test_a_configured_admin_token_is_used_without_a_login_round_trip(
+    tracking_transport,
+):
+    """Password login on every call would hammer the homeserver during a sweep."""
+    from shared.matrix import AdminCredentials
+
+    transport, requests = tracking_transport
+    client = SynapseAdminClient(
+        AdminCredentials(homeserver="https://matrix.example.com", admin_token="tok"),
```

Ruff (0.16.1)

[error] Possible hardcoded password assigned to argument: "admin_token"

(S106)